modules/add-contextual-data/contextual-data-record-scanner.c

```diff
@@ -103,26 +107,58 @@ _fetch_value(ContextualDataRecordScanner *self, ContextualDataRecord *record)
                   "to be escaped as '$$' once you change your @version declaration in the "
                   "configuration. This message means that this string is now assumed to be a "
                   "literal (non-template) string for compatibility",
+                  cfg_format_config_version_tag(self->cfg),
                   evt_tag_str("selector", record->selector->str),
                   evt_tag_str("name", log_msg_get_value_name(record->value_handle, NULL)),
                   evt_tag_str("value", value_template));
       log_template_compile_literal_string(record->value, value_template);
+      success = TRUE;
     }
-  else
+  else if (!cfg_is_typing_feature_enabled(self->cfg))
     {
-      GError *error = NULL;
-
-      if (!log_template_compile_with_type_hint(record->value, value_template, &error))
+      if (strchr(value_template, '(') != NULL)
         {
-          msg_error("add-contextual-data(): error compiling template",
-                    evt_tag_str("selector", record->selector->str),
-                    evt_tag_str("name", log_msg_get_value_name(record->value_handle, NULL)),
-                    evt_tag_str("value", value_template),
-                    evt_tag_str("error", error->message));
-          g_clear_error(&error);
-          return FALSE;
+          success = log_template_compile_with_type_hint(record->value, value_template, &error);
+          if (!success)
+            {
+              log_template_set_type_hint(record->value, "string", NULL);
+              msg_warning("WARNING: the value field in add-contextual-data() CSV files has been changed "
+                          "to support typing from " FEATURE_TYPING_VERSION ". You are using an older config "
+                          "version and your CSV file contains an unrecognized type-cast, probably a "
+                          "parenthesis in the value field. This will be interpreted in the `type(value)' "
+                          "format in future versions. Please add an "
+                          "explicit string() cast as shown in the 'fixed-value' tag of this log message "
+                          "or remove the parenthesis",
```

Comment:
Shouldn't we write something about the fact that although we throw a warning,we continue with `string` type hint?

Reply:
I added this to the warning.

Reply:
Thanks!